internal/provider/sdkv2/intercept.go

```diff
@@ -57,28 +58,18 @@ type interceptor1[D, E any] interface {
 	run(context.Context, interceptorOptions[D]) E
```

Comment:
Probably best to rename this to just `interceptor` as there are no longer `1` and `2` variants.